File: 안윤철/스택, 큐/같은숫자는싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        
        List<Integer> list = new ArrayList<>();
        list.add(arr[0]);
        
        for (int e: arr) {
            if (list.get(list.size()-1) != e) list.add(e);
        }
        
        int[] answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }
        
        
        return answer;
    }
}

File: 김금환/스택, 큐/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    
    if (arr.size() > 0)
        answer.push_back(arr[0]);
    
    for (int num : arr) {
        if (answer.back() != num)
            answer.push_back(num);
    }
    
    return answer;
}